src/cow-react/modules/mainMenu/state/mainMenuUrlOverridesAtom.ts

```diff
@@ -11,3 +11,11 @@ import { MainMenuItemId } from '../constants/mainMenu'
 export type MainMenuUrlOverrides = { [key in keyof Partial<typeof MainMenuItemId>]: string }
 
 export const mainMenuUrlOverridesAtom = atom<MainMenuUrlOverrides>({})
+
+export const updateMainMenuUrlOverridesAtom = atom(null, (get, set, nextState: Partial<MainMenuUrlOverrides>) => {
```

Comment:
I don't totally see this approach. Feels like overkill.

If I understand correctly:
- You have a hook to get the trade params
- Theres an updater that uses it
- The updater modifies the state
- The state flow down to `MenuTree` as a [list of overrides](https://github.com/cowprotocol/cowswap/blob/8ca383497895fdc7fbb04b8ca2d458a883cb8878/src/cow-react/modules/mainMenu/containers/MainMenu/index.tsx#L16)

For me it feels a bit of a overkill, and also complicates a bit the logic. 

I think i suggested it in the past, but i lost track of that conversation. If we are doing something more complex than a normal internal or external link, we have a system to create custom ones. 

In this case, we would just need to create a "smart link" that would use the hook you used in the updater.

Something like:
1.  We replace here https://github.com/cowprotocol/cowswap/blob/8ca383497895fdc7fbb04b8ca2d458a883cb8878/src/cow-react/modules/mainMenu/constants/mainMenu.ts#L91 `{ id: MainMenuItemId.SWAP, title: 'Swap', url: Routes.SWAP }` for something like `{ kind: MenuItemKind.SWAP_LINK },`
2. Of course we would need a new item here https://github.com/cowprotocol/cowswap/blob/8ca383497895fdc7fbb04b8ca2d458a883cb8878/src/cow-react/modules/mainMenu/constants/mainMenu.ts#L17 for the link
3. We make sure we represent the new link type here https://github.com/cowprotocol/cowswap/blob/8ca383497895fdc7fbb04b8ca2d458a883cb8878/src/cow-react/modules/mainMenu/pure/MenuTree/index.tsx#L98 
4. I think you will need some context to form the righ link https://github.com/cowprotocol/cowswap/blob/8ca383497895fdc7fbb04b8ca2d458a883cb8878/src/cow-react/modules/mainMenu/pure/MenuTree/index.tsx#L65 



For examp
https://github.com/cowprotocol/cowswap/blob/8ca383497895fdc7fbb04b8ca2d458a883cb8878/src/cow-react/modules/mainMenu/pure/MenuTree/index.tsx#L99
5. This params from the context needs to be injected from here https://github.com/cowprotocol/cowswap/blob/8ca383497895fdc7fbb04b8ca2d458a883cb8878/src/custom/components/Header/index.tsx#L108

Reply:
Thank you!
Fixed: https://github.com/cowprotocol/cowswap/pull/1270/commits/be0f14cc1c71aefc7b065a1366face4045387743